src/parser.rs

```diff
@@ -243,13 +259,13 @@ assert_eq!(
 
         self.next();
 
-        while self.peek().is_some() {
+        loop {
             self.consume_sp_chars();
 
-            if Some(b')') == self.peek() {
-                self.next();
-                let param_visitor = visitor.finish().map_err(Error::custom)?;
-                return self.parse_parameters(param_visitor);
+            match self.peek() {
```

Comment:
I refactored this loop to move the error check inside and the (now) more complex parameter processing to the mainline.